src/wtmp.c

```diff
@@ -40,7 +40,7 @@ int log_wtmp(const char *line, const char *name, const char *host,
     !(defined(LINUX) || defined(__hpux) || defined (_AIX))
   /* This "auxilliary" utmp doesn't exist under linux. */
 
-#if defined(__sparcv9) && !defined(__NetBSD__) && !defined(__FreeBSD__)
+#if defined(__sun)
```

Comment:
Ok. What I meant to say is that I think this code path was never meant for any other OS than Solaris, because `struct futmpx` is not defined anywhere else.
My main goal is to make code correct, but my secondary goal is also to make nicer/simpler code, if it works the same.
I'll update the pull request.